runtime/parser/statement.go

```diff
@@ -94,7 +94,7 @@ func parseStatement(p *parser) (ast.Statement, error) {
 			return parseForStatement(p)
 		case keywordEmit:
 			return parseEmitStatement(p)
-		case keywordFun:
+		case keywordFun, keywordView:
```

Comment:
Same as above. Add a separate case for the `view` keyword and ensure the `fun` keyword follows.
Like above, add a purity parameter to `parseFunctionDeclarationOrFunctionExpressionStatement`